pkg/kubelet/lifecycle/predicate.go

```diff
@@ -161,6 +161,14 @@ func (w *predicateAdmitHandler) Admit(attrs *PodAdmitAttributes) PodAdmitResult
 			Message: "Failed to admit pod as the `kubernetes.io/os` label doesn't match node label",
 		}
 	}
+	// By this time, node labels should have been synced, this helps in identifying the pod with the usage.
+	if rejectPodAdmissionBasedOnOSField(admitPod) {
+		return PodAdmitResult{
+			Admit:   false,
+			Reason:  "PodOSFieldNodeLabelDoesNotMatch",
```

Comment:
drop NodeLabel from the reason… we're not comparing to the label

Reply:
pick a reason that will continue to make sense if LCOW is ever supported… maybe `PodOSNotSupported` or something?

Reply:
The only reason for keeping it is - the user may not know about the OS of the node. Sometimes the only object he/she may have access to is node object.